src/main/java/budgetbuddy/model/person/Person.java

```diff
@@ -102,6 +86,12 @@ public boolean hasLoan(Loan loan) {
     public void addLoan(Loan loan) {
         requireNonNull(loan);
         loans.add(loan);
+        for (Loan oldLoan : loans) {
```

Comment:
Is this maybe in the wrong place? I don't think adding a loan requires a for loop with "oldLoan"s?

Reply:
Er yes, I have to admit it's a bit strange, but it's in the right place. Each loan in the person's list has a reference to a (identical) person, which becomes outdated once the loan is added to the actual person's loan list. So the "old" loans need to be updated with the new person. This is a sad result of the `person` reference in each loan being final... To solve this, perhaps I could make it not final in a future PR?